testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/metrics/metadata/MicroProfileMetricsMetadataTestCase.java

```diff
@@ -49,18 +50,28 @@
     @ContainerResource
     ManagementClient managementClient;
 
+    @Test
+    public void testJsonHeader() throws Exception {
+        final String endpointURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/metrics";
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpOptions request = new HttpOptions(endpointURL);
+            request.setHeader("Accept", String.valueOf(ContentType.APPLICATION_JSON));
+            CloseableHttpResponse resp = client.execute(request);
+            assertEquals(200, resp.getStatusLine().getStatusCode());
+        }
+    }
+
     @Test
     public void testNoJsonHeader() throws Exception {
         final String endpointURL = "http://" + managementClient.getMgmtAddress() + ":" + managementClient.getMgmtPort() + "/metrics";
         try (CloseableHttpClient client = HttpClients.createDefault()) {
 
             CloseableHttpResponse resp = client.execute(new HttpOptions(endpointURL));
-
             assertEquals(406, resp.getStatusLine().getStatusCode());
             String content = EntityUtils.toString(resp.getEntity());
             resp.close();
-            assertTrue("'No exporter found for method OPTIONS and media type' message is expected, but was: " + content,
-                    content.contains("No exporter found for method OPTIONS and media type"));
+            assertTrue("'OPTIONS method is only allowed with application/json media type.' message is expected, but was: " + content,
```

Comment:
Having the test depending on the text is a bit brittle. Checking that the HTTP request returns 406 should be enough

Reply:
Well, the test was already depending on the text before this change and I updated it to check https://issues.jboss.org/browse/WFWIP-151 so I think the text is important here.

Reply:
+1 to just the 406. We should not be asserting strings provided by components that have no requirement to keep the same string.

Reply:
Those string assertions were added by me 3 months ago and nobody objected.
I expect similar string checks are performed at many TestCases.

I think there should be separate issue to revisit all the MP components related checks.
For now I would not drop the check.